qiskit_experiments/test/t2hahn_backend.py

```diff
@@ -13,337 +13,206 @@
-        }
-        for circ in run_input:
-            nqubits = circ.num_qubits
-            qubit_indices = {bit: idx for idx, bit in enumerate(circ.qubits)}
-            clbit_indices = {bit: idx for idx, bit in enumerate(circ.clbits)}
-            counts = dict()
-
-            for _ in range(shots):
-                qubit_state = self._qubit_initialization(
-                    nqubits=nqubits
-                )  # for parallel need to make an array
-                clbits = np.zeros(circ.num_clbits, dtype=int)
-                for op, qargs, cargs in circ.data:
-                    qubit = qubit_indices[qargs[0]]
-
-                    # The noise will only be applied if we are in the XY plane.
-                    if op.name == "delay":
-                        delay = op.params[0]
-                        t2hahn = self._t2hahn[qubit]
-                        freq = self._frequency[qubit]
-                        qubit_state[qubit] = self._delay_gate(
-                            qubit_state=qubit_state[qubit],
-                            delay=delay,
-                            t2hahn=t2hahn,
-                            frequency=freq,
-                        )
-                    elif op.name == "rx":
-                        qubit_state[qubit] = self._rx_gate(qubit_state[qubit], op.params[0])
-                    elif op.name == "measure":
-                        meas_res = self._measurement_gate(qubit_state[qubit])
-                        clbit = clbit_indices[cargs[0]]
-                        clbits[clbit] = meas_res
-
-                clstr = ""
-                for clbit in clbits[::-1]:
-                    clstr = clstr + str(clbit)
-
-                if clstr in counts:
-                    counts[clstr] += 1
-                else:
-                    counts[clstr] = 1
-            result["results"].append(
-                {
-                    "shots": shots,
-                    "success": True,
-                    "header": {"metadata": circ.metadata},
-                    "data": {"counts": counts},
-                }
-            )
-        return FakeJob(self, Result.from_dict(result))
+        noise_model = NoiseModel()
+
+        if self._initialization_error is not None:
+            passes.append(ResetQubits())
+            for qubit, error in enumerate(self._initialization_error):
+                if error is None:
+                    continue
+                noise_model.add_quantum_error(reset_error(1 - error, error), ["reset"], [qubit])
+
+        readout0to1 = self._readout0to1
+        readout1to0 = self._readout1to0
+        if readout0to1 and not readout1to0:
+            readout1to0 = [0] * self.num_qubits
+        elif readout1to0 and not readout0to1:
+            readout0to1 = [0] * self.num_qubits
+        if readout0to1:
+            for qubit, (err0to1, err1to0) in enumerate(zip(readout0to1, readout1to0)):
+                if err0to1 is None:
+                    err0to1 = 0
```

Comment:
```suggestion
                err0to1 = err0to1 or 0
```

Reply:
While starting to address this, I decided that it was not necessary to support `None` arguments so much, so I removed support for them in 3c49a98e7a728f9f4fe4f3312300444c1a2dcd0c.